packages/dds/tree/src/shared-tree-core/commitEnricher.ts

```diff
@@ -0,0 +1,41 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { GraphCommit } from "../core/index.js";
+
+/**
+ * A component that can be used to enrich outbound commits.
+ */
+export interface CommitEnricher<TChange> {
+	/**
+	 * Enriches a commit with adequate refreshers.
+	 * Assumes that the commit will be submitted.
+	 * @param commit - the commit to enrich. Not mutated.
+	 * @param isResubmit - whether the commit is a resubmission of a previously submitted commit.
+	 * Must be true if and only if `isInResubmitPhase` is true.
+	 * @returns the enriched commit. Possibly the same as the one passed in.
+	 */
+	enrichCommit(commit: GraphCommit<TChange>, isResubmit: boolean): GraphCommit<TChange>;
+
+	/**
+	 * Must be called at the start of a resubmission phase before calling `enrichCommit`.
+	 * @param toResubmit - the commits that will be resubmitted (from oldest to newest).
+	 * This must be the most rebased version of these commits (i.e., rebased over all known concurrent edits)
+	 * as opposed to the version which was last submitted.
+	 * Can be safely mutated after this call returns.
+	 */
+	startResubmitPhase(toResubmit: readonly GraphCommit<TChange>[]): void;
```

Comment:
I agree it's a smidge smelly. Ideally this method would be the only thing that needs to be called for the resubmit phase (in which case this method wouldn't exist and `enrichCommit` would take an array instead), we're limited by `SharedObject`'s choice of having the resubmit phase happen one commit at a time instead of as a batch.

I agree we can do better on the name. I think `processResubmit` runs the risk sounding like the whole resubmit business is taken care of once this function returns. I propose `prepareForResubmit`.